### PR TITLE
wrapped in vertx thread model

### PR DESCRIPTION
**Pull request by fmarinelli**
Now it's possible to execute tests with check threads on

### CODE REVIEW

File: src/main/java/io/vertx/java/spi/cluster/impl/jgroups/services/DefaultRpcExecutorService.java [hunk 1]
```diff
@@ -80,13 +79,16 @@ public DefaultRpcExecutorService(RpcDispatcher dispatcher) {
...
       NotifyingFuture<RspList<T>> notifyingFuture = this.<T>execute(action, options);
       notifyingFuture.setListener((future) -> {
-        try {
-          RspList<T> rspList = future.get();
-          T result = futureDone(rspList);
-          handler.handle(Future.succeededFuture(result));
-        } catch (Exception e) {
-          handler.handle(Future.failedFuture(e));
-        }
+        vertx.executeBlocking(
```
Comment (purplefox) [inline src/main/java/io/vertx/java/spi/cluster/impl/jgroups/services/DefaultRpcExecutorService.java]: I don't understand this - isn't the whole point of a future listener that it's not called until the result is ready? So why use executeBlocking for the get()?

Comment (fmarinelli) [inline src/main/java/io/vertx/java/spi/cluster/impl/jgroups/services/DefaultRpcExecutorService.java]: to execute the callback in vertx threads

Comment (fmarinelli) [inline src/main/java/io/vertx/java/spi/cluster/impl/jgroups/services/DefaultRpcExecutorService.java]: I can change that with runOnContext.
what do you think?

Comment (purplefox) [inline src/main/java/io/vertx/java/spi/cluster/impl/jgroups/services/DefaultRpcExecutorService.java]: To execute a callback on the correct thread you should use context.runOnContext(). executeBlocking is for running blocking code.

Comment (fmarinelli) [inline src/main/java/io/vertx/java/spi/cluster/impl/jgroups/services/DefaultRpcExecutorService.java]: ok, I'll change that asap.

**Comment by purplefox** [comment]
Fabio - you should go and enjoy your holiday :)

There is no rush to fix this today!

**Pull request merged**